community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/PopulatingIndexProxy.java

```diff
@@ -80,10 +81,13 @@ public IndexUpdater newUpdater( final IndexUpdateMode mode )
         return new IndexUpdater()
         {
             @Override
-            public void process( NodePropertyUpdate update ) throws IOException, IndexEntryConflictException
```

Comment:
This doesn't seem right. At the `prepare` the transaction is not yet committed, so applying it to the store would be wrong, and this method forwards the update straight to the index populator, i.e. applies the update to the index before it has been committed. That is not right, it should return a `PreparedIndexUpdates` object that performs that task on `commit()`